components/core/src/crypto/keys/box_key_pair.rs

```diff
@@ -563,12 +622,41 @@ mod test {
             .unwrap();
         let user = BoxKeyPair::generate_pair_for_user("wecoyote", cache.path()).unwrap();
 
-        let ciphertext = service.encrypt("Out of rockets".as_bytes(), &user).unwrap();
+        let ciphertext = service
+            .encrypt("Out of rockets".as_bytes(), Some(&user))
+            .unwrap();
         let message = BoxKeyPair::decrypt(&ciphertext, cache.path()).unwrap();
         assert_eq!(message, "Out of rockets".as_bytes());
     }
 
     #[test]
+    fn encrypt_and_decrypt_to_self() {
+        let cache = TempDir::new("key_cache").unwrap();
+        let sender = BoxKeyPair::generate_pair_for_user("wecoyote", cache.path()).unwrap();
+
+        let ciphertext = sender.encrypt("Buy more rockets".as_bytes(), None).unwrap();
+        let message = BoxKeyPair::decrypt(&ciphertext, cache.path()).unwrap();
+        assert_eq!(message, "Buy more rockets".as_bytes());
+    }
+
+    #[test]
+    fn encrypt_to_self_with_only_public_key() {
+        let cache = TempDir::new("key_cache").unwrap();
+        let sender = BoxKeyPair::generate_pair_for_user("wecoyote", cache.path()).unwrap();
+
+        // Delete the sender's secret key
+        fs::remove_file(
+            BoxKeyPair::get_secret_key_path(&sender.name_with_rev(), cache.path()).unwrap(),
+        ).unwrap();
+        // Now reload the sender's pair which will be missing the secret key
+        let sender = BoxKeyPair::get_latest_pair_for("wecoyote", cache.path()).unwrap();
+
+        sender
+            .encrypt("Nothing to see here".as_bytes(), None)
+            .unwrap();
+    }
```

Comment:
Kind of weird to see a test with no assertions... could this be rewritten to use assertions so it's clear what exactly is the crux of the test?